### PR TITLE
feat(unstable): support in memory certificate data for creating httpClient

### PR DESCRIPTION
**Pull request by yonbav**
- Add unit test that recieve certificate data as parameter
- Add caData to CreateHttpClientOptions
- fetch/lib.rs: create the certificate from ca_data

issue: https://github.com/denoland/deno/issues/8190

### CODE REVIEW

**Comment by bartlomieju** [comment]
@ry @lucacasonato I think this PR is great and I'd prefer this approach to taking a path to CA file. So I suggest we land this one and remove `caPath` in a follow up PR, that way `deno_fetch` needn't be aware of checking path permission.

File: cli/dts/lib.deno.unstable.d.ts [hunk 1]
```diff
@@ -1266,6 +1266,7 @@ declare namespace Deno {
      * Requires `allow-read` permission.
      */
     caFile?: string;
+    caData?: string;
```
Comment (lucacasonato) [inline cli/dts/lib.deno.unstable.d.ts:1269]: Can you add a JSDoc comment here (this doesn't need allow-read). Also could you mention this takes a PEM encoded certificate?

File: op_crates/fetch/lib.rs [hunk 2]
```diff
@@ -263,13 +266,20 @@ where
...
+  ca_file: Option<&str>,
+  ca_data: Option<&str>,
+) -> Result<Client, AnyError> {
   let mut builder = Client::builder().redirect(Policy::none()).use_rustls_tls();
   if let Some(ca_file) = ca_file {
     let mut buf = Vec::new();
     File::open(ca_file)?.read_to_end(&mut buf)?;
     let cert = reqwest::Certificate::from_pem(&buf)?;
     builder = builder.add_root_certificate(cert);
+  } else if let Some(ca_data) = ca_data {
```
Comment (lucacasonato) [inline op_crates/fetch/lib.rs:279]: `ca_data` should have priority over `ca_file`.

**Review by lucacasonato — COMMENTED** [review]
Very nice. I agree with removing caPath in a follow-up PR. Just a few comments.

File: cli/dts/lib.deno.unstable.d.ts [hunk 3]
```diff
@@ -1266,6 +1266,12 @@ declare namespace Deno {
      * Requires `allow-read` permission.
      */
     caFile?: string;
+
+    /** A certificate authority to use when validating TLS certificates.
+     *
+     * Pem format certificate data
```
Comment (lucacasonato) [inline cli/dts/lib.deno.unstable.d.ts:1272]:
```suggestion
    /** A certificate authority to use when validating TLS certificates. Certificate data must be PEM encoded.
```

**Review by lucacasonato — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**